web/src/engine/websites/GalaxyManga.ts

```diff
@@ -10,20 +11,42 @@ function ChapterInfoExtractor(element: HTMLAnchorElement) {
     };
 }
 
-@Common.MangaCSS(/^{origin}\/series\/\d+-\d+-[^/]+$/, 'main section > div > div.font-semibold')
 @Common.ChaptersSinglePageCSS('main section div.overflow-y-auto div.grid > a', ChapterInfoExtractor)
 @Common.PagesSinglePageCSS('main > section > div > img')
 @Common.ImageAjax()
 export default class extends DecoratableMangaScraper {
+    private domainUpdated = false;
+    private readonly mangapattern = /^{origin}\/series\/\d+-\d+-[^/]+$/;
 
     public constructor() {
-        super('galaxymanga', 'Galaxy Manga', 'https://galaxymanga.net', Tags.Media.Manga, Tags.Media.Manhwa, Tags.Media.Manhua, Tags.Language.Arabic, Tags.Source.Scanlator);
+        super('galaxymanga', 'Galaxy Manga', 'https://outdatedwebsite.net', Tags.Media.Manga, Tags.Media.Manhwa, Tags.Media.Manhua, Tags.Language.Arabic, Tags.Source.Scanlator);
     }
 
     public override get Icon() {
         return icon;
     }
 
+    private async DomainUpdate(): Promise<void> {
+        if (this.domainUpdated) return;
+        const [element] = await FetchCSS(new Request('https://gxcomic.xyz'), 'div.container p strong');
+        this.URI.href = element ? element.textContent.startsWith('https://') ? element.textContent.trim() : `https://${element.textContent.trim()}` : this.URI.href;
+        this.domainUpdated = true;
+    }
+
+    public override async Initialize(): Promise<void> {
+        await this.DomainUpdate();
+    }
+
+    public override async ValidateMangaURL(url: string): Promise<boolean> {
+        await this.DomainUpdate();
```

Comment:
You are totally right, and i thought about that already.

For now there is one website that dont follow any regex pattern and need to fetch from a portal page, perhaps two or three at **maximum**. Since its only done once and on a couple of website i deemed this workaround acceptable .

Feel free to close this PR in case my idea was just silly :D